src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -177,7 +180,8 @@ private async Task RunStepsAsync(List<IStep> compositeSteps)
                 var actionStep = step as IActionRunner;
 
                 // Set GITHUB_ACTION
-                step.ExecutionContext.SetGitHubContext("action", step.ExecutionContext.GetFullyQualifiedContextName());
+                // For composite + their nested steps, we want the all of them to have the same GITHUB_ACTION
+                step.ExecutionContext.SetGitHubContext("action", ExecutionContext.GetGitHubContext("action"));
```

Comment:
can we delete this instead?

Reply:
Yes, we can just delete this. I double checked and printed the `step.ExecutionContext.ExpressionValues["github"]` at this location and it has the same github.action value as expected.